sycl/include/CL/sycl/id.hpp

```diff
@@ -75,486 +75,77 @@ template <int dimensions = 1> struct id : public detail::array<dimensions> {
-      result.common_array[i] = this->common_array[i] || rhs;
-    }
-    return result;
-  }
-  id<dimensions> operator<(const size_t &rhs) const {
-    id<dimensions> result;
-    for (int i = 0; i < dimensions; ++i) {
-      result.common_array[i] = this->common_array[i] < rhs;
-    }
-    return result;
-  }
-  id<dimensions> operator>(const size_t &rhs) const {
-    id<dimensions> result;
-    for (int i = 0; i < dimensions; ++i) {
-      result.common_array[i] = this->common_array[i] > rhs;
-    }
-    return result;
-  }
-  id<dimensions> operator<=(const size_t &rhs) const {
-    id<dimensions> result;
-    for (int i = 0; i < dimensions; ++i) {
-      result.common_array[i] = this->common_array[i] <= rhs;
-    }
-    return result;
-  }
-  id<dimensions> operator>=(const size_t &rhs) const {
-    id<dimensions> result;
-    for (int i = 0; i < dimensions; ++i) {
-      result.common_array[i] = this->common_array[i] >= rhs;
-    }
-    return result;
-  }
-
-  // OP is: +=, -=, *=, /=, %=, <<=, >>=, &=, |=, ^=
-  id<dimensions> &operator+=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] += rhs[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator-=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] -= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator*=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] *= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator/=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] /= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator%=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] %= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator<<=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] <<= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator>>=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] >>= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator&=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] &= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator|=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] |= rhs.common_array[i];
-    }
-    return *this;
-  }
-  id<dimensions> &operator^=(const id<dimensions> &rhs) {
-    for (int i = 0; i < dimensions; ++i) {
-      this->common_array[i] ^= rhs.common_array[i];
-    }
-    return *this;
-  }
+  #define GEN_OP(op)                                                           \
```

Comment:
Using such small macro names might cause some conflicts with some other libraries or user code.
What about using some reserved names, such as `__SYCL_GEN_OPT`? 
Note that this comment does apply to some other macros in some other parts of the current implementation that should be changed before up-streaming...